ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezTask.java

```diff
@@ -236,6 +239,10 @@ public int execute() {
           throw new HiveException("Operation cancelled");
         }
 
+        // Log all the info required to find the various logs for this query
+        LOG.info("HS2 Host: [{}], Query ID: [{}], Dag ID: [{}], DAG Session ID: [{}]", getHostNameIP(), queryId,
```

Comment:
I would change that to something like "Executing query with Id {} from dag {} by session {} "
Not sure I would add the hostName IP here as this can be quite confusing, especially when running mutiple containers on the same node (like kubernetes). Not sure if we already have better info for that on a Task level.

Reply:
Hey @pgaref, thanks for the valuable input.

So, on one hand it may seem confusing as to the need to log the host name because if you are looking at this log file then of course the hostname is already known.  However, as I understand the code, these log messages are sent  (redirected) to the client via Thrift RPC APIs.  This is helpful information for debugging for the client as it's not otherwise clear which instance of HS2 is processing the query, if for example, a load balancer is between the client and HS2.  I had thought about including some sort of HS2 unique ID as well, but I didn't find such a capability in the project already and did not want to introduce here.

However, these 4 pieces of information allow a client to report a problem to the admin and allow the admin to grab all the log files: HS2 and YARN TEZ DAG logs.

Reply:
@pgaref Have you had a chance to think about this topic?  What do you think about getting this in for the on-prem solutions and then modify in a subsequent ticket for multi-HS2 hosts?

Reply:
Hey @belugabehr  -- taking another look here and seems like SessionState could do the job.
What about SessionState.getHiveServer2Host() ?

Reply:
I'd hate to tie anything to the session state.
```java
      // Need to remove this static hack. But this is the way currently to get a session.
      SessionState ss = SessionState.get();
```

Let me see what else we can do.

Reply:
@pgaref I at least re-used the `hive-common` package feature here.  Please review once more.  Let not the perfect be the enemy of the good.